File: scrimage-core/src/test/scala/com/sksamuel/scrimage/LinearSubpixelInterpolatorTest.scala
package com.sksamuel.scrimage

import org.scalatest.FunSuite

class LinearSubpixelInterpolatorTest extends FunSuite {

  private val image = ImmutableImage.fromResource("/com/sksamuel/scrimage/bird.jpg")

  test("subpixel happy path") {
    assert(-1315602 === image.subpixel(2, 3))
    assert(-1381395 === image.subpixel(22, 63))
    assert(-2038553 === image.subpixel(152, 383))
  }

  test("subimage has right dimensions") {
    val covered = image.cover(200, 200)
    val subimage = covered.subimage(10, 10, 50, 100)
    assert(50 === subimage.width)
    assert(100 === subimage.height)
  }
}

File: scrimage-core/src/test/kotlin/com/sksamuel/scrimage/core/subpixel/LinearSubpixelInterpolatorTest.kt
package com.sksamuel.scrimage.core.subpixel

import com.sksamuel.scrimage.ImmutableImage
import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldBe

class LinearSubpixelInterpolatorTest : FunSpec({

   val image = ImmutableImage.fromResource("/com/sksamuel/scrimage/bird.jpg")

   test("subpixel happy path") {
      -1315602 shouldBe image.subpixel(2.0, 3.0)
      -1381395 shouldBe image.subpixel(22.0, 63.0)
      -2038553 shouldBe image.subpixel(152.0, 383.0)
   }

   test("subimage has right dimensions") {
      val covered = image.cover(200, 200)
      val subimage = covered.subimage(10, 10, 50, 100)
      50 shouldBe subimage.width
      100 shouldBe subimage.height
   }

})
